pkg/operator/controller/route-status/controller_test.go

```diff
@@ -0,0 +1,197 @@
+	"testing"
+)
+
+// Test_Reconcile verifies Reconcile
+// Note: This effectively tests clearStaleRouteAdmittedStatus as well
+func Test_Reconcile(t *testing.T) {
+	reconcileRequestRoute := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "route",
+			Namespace: "ns",
+		},
+	}
+	testCases := []struct {
+		name              string
+		request           reconcile.Request
+		route             *routev1.Route
+		ingressController *v12.IngressController
+		namespace         *corev1.Namespace
+		expectedRoute     *routev1.Route
+		expectedErr       bool
+	}{
+		{
+			name:              "admitted with no routeSelector and no namespaceSelector with route with no labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "admitted with no routeSelector and no namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "admitted with no routeSelector and no namespaceSelector with route with no labels in namespace with correct labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", "shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "admitted with routeSelector and no namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "shard-label", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "shard-label", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "not admitted with routeSelector and no namespaceSelector with route with no labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
```

Comment:
It looks like the route's status reports that it *was* admitted, right?  Should I read "not admitted" as "stale Admitted status condition"?